### PR TITLE
Migrate `issue-37839`, `track-path-dep-info` and `track-pgo-dep-info` `run-make` tests to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

Please try:

try-job: test-various
try-job: dist-x86_64-musl

### CODE REVIEW

**Comment by jieyouxu** [comment]
Record: passed on `test-various` and `dist-x86_64-musl`.

File: tests/run-make/track-pgo-dep-info/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+//FIXME(Oneirical): try it on musl
+//@ needs-profiler-support
+
+use run_make_support::{fs_wrapper, llvm_profdata, run, rustc};
+
+fn main() {
+    // Generate the profile-guided-optimization (PGO) profiles
+    rustc().profile_generate("profiles").input("main.rs").run();
+    // Merge the profiles
+    run("main");
```
Comment (jieyouxu) [inline tests/run-make/track-pgo-dep-info/rmake.rs:16]: Question: does this need `//@ ignore-cross-compile`? Or is that kinda implied by `//@ needs-profiler-support` (even if it is, maybe having the ignore is clearer?)?

**Review by jieyouxu — APPROVED** [review]
Looks good, just one question about cross-compile case.

File: tests/run-make/track-pgo-dep-info/rmake.rs (hunk 1 above)
Comment (Oneirical) [reply tests/run-make/track-pgo-dep-info/rmake.rs:17]: I believe that it's indeed getting implied by //@ needs-profiler-support, but I added the ignore + reason for clarity.

**Comment by jieyouxu** [comment]
Thanks!

@bors r+ rollup=iffy (profdata shenanigans)

**Pull request merged**